src/hooks/useEditorWordWrapSetting.ts

```diff
@@ -0,0 +1,33 @@
+import { EditorSettings } from '@/types/settings'
+import { useState, useEffect } from 'react'
+import { useDebounce } from 'react-use'
+
+export function useEditorWordWrapSetting(key: keyof EditorSettings) {
```

Comment:
This hook can be used to handle each setting independently based on the `key` (`wordWrapScriptPreview` or `wordWrapResponseContent`)